hi-ml-histopathology/src/histopathology/models/deepmil.py

```diff
@@ -260,24 +260,25 @@ def training_step(self, batch: Dict, batch_idx: int) -> Tensor:  # type: ignore
                  sync_dist=True)
         if self.verbose:
             print(f"After loading images batch {batch_idx} -", _format_cuda_memory_stats())
-        self.log_metrics(ModelKey.TRAIN)
         return train_result[ResultsKey.LOSS]
 
     def validation_step(self, batch: Dict, batch_idx: int) -> BatchResultsType:  # type: ignore
         val_result = self._shared_step(batch, batch_idx, ModelKey.VAL)
         self.log('val/loss', val_result[ResultsKey.LOSS], on_epoch=True, on_step=True, logger=True,
                  sync_dist=True)
-        self.log_metrics(ModelKey.VAL)
         return val_result
 
     def test_step(self, batch: Dict, batch_idx: int) -> BatchResultsType:  # type: ignore
         test_result = self._shared_step(batch, batch_idx, ModelKey.TEST)
         self.log('test/loss', test_result[ResultsKey.LOSS], on_epoch=True, on_step=True, logger=True,
                  sync_dist=True)
-        self.log_metrics(ModelKey.TEST)
         return test_result
 
+    def training_epoch_end(self, outputs: EpochResultsType) -> None:
+        self.log_metrics(ModelKey.TRAIN)
+
     def validation_epoch_end(self, epoch_results: EpochResultsType) -> None:  # type: ignore
+        self.log_metrics(ModelKey.VAL)
```

Comment:
Suggestion, replace `VAL` with `VALIDATION`.

Reply:
This would need changing at multiple places, perhaps could be addressed in a separate PR.